devito/data/data.py

```diff
@@ -41,7 +41,7 @@ class Data(np.ndarray):
     def __new__(cls, shape, dtype, decomposition=None, modulo=None, allocator=ALLOC_FLAT):
         assert len(shape) == len(modulo)
         ndarray, memfree_args = allocator.alloc(shape, dtype)
-        obj = np.asarray(ndarray).view(cls)
+        obj = ndarray.view(cls)
```

Comment:
curiosity: is this just aesthetics or is it changing behaviour ?

Reply:
Yes and No. 
The allocation procedure in allocator should have already done the appropriate wrapping-as-numpy-array (hence the name of the variable being `ndarray`) - in which case, `np.asarray` is not required. From this point of view it is just aesthetics. 
However, if the allocator has done something "special" (in this case, setting some flags for numpy), this effectively undoes that and redoes the wrapping. From this point of view it is changing behaviour.